cloud: add a new article about naming conventions for data import (#10990)
pingcap/docs: tidb-cloud/naming-conventions-for-data-import.md

---
title: Naming Conventions for Data Import
summary: Learn about the requirements of file formats for data import.
---

# Naming Conventions for Data Import

The Data Import feature in TiDB Cloud supports the following file formats: CSV, Parquet, Aurora Snapshot, and SQL File. There are two prerequisites:

- The data files you provide and the schema files for building the database and tables must conform to the naming requirements. If you do not provide the schema file for building the database, TiDB Cloud will create the target database for you according to the default configuration. If you cannot provide data files with the required file names, you can modify the file pattern to perform the import task. Otherwise, TiDB Cloud can not scan the files you need to import.
- If you can not provide the schema file for building tables, you need to create the target tables in TiDB Cloud in advance.

## Naming conventions for data files and schema files

This section describes the naming conventions for data files and schema files.

> 

Reviewer: Since the schema file naming are the same for all 4 types of data sources, it is recommended to provide a separate section to describe the schema file naming, with examples.

Author: Sounds good. Updated.

### CSV

When you import CSV files, name the schema files and data files as follows:

- DB schema file (optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`

> - Data file: `${db_name}.${table_name}[.XXXXXX].csv` ([.XXXXXX] is optional)

Author: OK updated.

> For example, `import_db.test_table.01.csv`

Reviewer: ```suggestion
For example, `import_db.test_table.csv`, `import_db.test_table.01.csv`
```

### Parquet

When you import Parquet files, name the schema files and data files as follows:

> - DB schema file(optional): `${db_name}-schema-create.sql`

Reviewer: ```suggestion
- DB schema file (optional): `${db_name}-schema-create.sql`
```

- Table schema file: `${db_name}.${table_name}-schema.sql`

> - Data file: `${db_name}.${table_name}[.XXXXXX].{snappy|gz|lzo}.parquet` (`[.XXXXXXX].{snappy|gz|lzo}` is optional )

Reviewer: ```suggestion
- Data file: `${db_name}.${table_name}[.XXXXXX][.{snappy|gz|lzo}].parquet` (`[.XXXXXXX]` and `[.{snappy|gz|lzo}]` are optional )
```

> For example, `import_db.test_table.01.parquet`

Reviewer: ```suggestion
For example, `import_db.test_table.parquet`, `import_db.test_table.01.parquet`, 
 `import_db.test_table.gz.parquet`, `import_db.test_table.01.gz.parquet`
```

### Aurora Snapshot

When you import Aurora Snapshot files, name the schema files and data files as follows:

> - DB schema file: `${db_name}-schema-create.sql (optional)`

Reviewer: ```suggestion
- DB schema file (optional): `${db_name}-schema-create.sql`
```

- Table schema file: `${db_name}.${table_name}-schema.sql`

> - Data file: All files with the `.parquet` suffix in the `db_name.table_name/` folder.

Reviewer: ```suggestion
- Data file: All files with the `.parquet` suffix in the `${db_name}.${table_name}/` folder.
```

> 

Reviewer 2: Any prefix consisting of "a-z, 0-9, - , _ , ." and suffix ".parquet" is fine, such as dbname.tablename.01.parquet.

Author: Add this sentence. PTAL.

### SQL

When you import SQL files, name the schema files and data files as follows:

> - DB schema file: `${db_name}-schema-create.sql` (optional)

Reviewer: ```suggestion
- DB schema file (optional): `${db_name}-schema-create.sql`
```

- Table schema file: `${db_name}.${table_name}-schema.sql`
- Data file: `${db_name}.${table_name}[.XXXXXXX].sql` ([.XXXXXXX] is optional)

> For example, `import_db.test_table.01.sql`

Reviewer: ```suggestion
For example, `import_db.test_table.sql`, `import_db.test_table.01.sql`
```

If the SQL file is exported through TiDB Dumpling with the default configuration, it conforms to the naming convention by default.

## File pattern

> If the source file of CSV or Parquet does not conform to the naming convention, you can use File pattern to establish the mapping relationship between the source file and the target table, so as to realize the data import. This feature does not support Aurora Snapshot and SQL File.

Reviewer 2: "the mapping relationship between the source file and the target table "  changes to  "the name mapping relationship between the source  data file and the target table"

This section describes the naming conventions for data files and schema files.

### CSV

When you import CSV files, name the schema files and data files as follows:

- DB schema file (optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`
- Data file: `${db_name}.${table_name}[.XXXXXX].csv` ([.XXXXXX] is optional)

For example, `import_db.test_table.csv`, `import_db.test_table.01.csv`

### Parquet

When you import Parquet files, name the schema files and data files as follows:

- DB schema file (optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`
- Data file: `${db_name}.${table_name}[.XXXXXX][.{snappy|gz|lzo}].parquet` (`[.XXXXXXX]` and `[.{snappy|gz|lzo}]` are optional )

For example, `import_db.test_table.parquet`, `import_db.test_table.01.parquet`, 
 `import_db.test_table.gz.parquet`, `import_db.test_table.01.gz.parquet`

### Aurora Snapshot

When you import Aurora Snapshot files, name the schema files and data files as follows:

- DB schema file (optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`

> - Data file: All files with the `.parquet` suffix in the `${db_name}.${table_name}/` folder.

Author: ```suggestion
- Data file: All files with the `.parquet` suffix in the `${db_name}.${table_name}/` folder. Any prefix consisting of "a-z, 0-9, - , _ , ." and suffix ".parquet" is fine, such as `dbname.tablename.01.parquet`.
```

### SQL

When you import SQL files, name the schema files and data files as follows:

- DB schema file (optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`
- Data file: `${db_name}.${table_name}[.XXXXXXX].sql` ([.XXXXXXX] is optional)

For example, `import_db.test_table.sql`, `import_db.test_table.01.sql`

If the SQL file is exported through TiDB Dumpling with the default configuration, it conforms to the naming convention by default.

## File pattern

> If the source file of CSV or Parquet does not conform to the naming convention, you can use File pattern to establish the mapping relationship between the source file and the target table, so as to realize the data import. This feature does not support Aurora Snapshot and SQL File.

Author: ```suggestion
If the source file of CSV or Parquet does not conform to the naming convention, you can use File pattern to establish the name mapping relationship between the source data file and the target table, so as to perform the data import task. This feature does not support Aurora Snapshot and SQL File.
```

---
title: Naming Conventions for Data Import
summary: Learn about the naming conventions of CSV, Parquet, Aurora Snapshot, and SQL files.
---

# Naming Conventions for Data Import

The Data Import feature in TiDB Cloud supports the following file formats: CSV, Parquet, Aurora Snapshot, and SQL. To make sure that data can be imported successfully, note the following:

> - The data files and the schema files for building the database and tables must conform to the naming conventions.
> - If you can not provide the schema file for building the database, TiDB Cloud will create a target database for you according to the default configuration.

Author: ```suggestion
- Prepare the SQL file for creating the target database (optional) and the SQL file for creating the target table that conform to the naming conventions. If the SQL file for creating the target table is not provided, you need to create the corresponding table manually in the target database in advance.
- Prepare a data file that conforms to the naming conventions for importing data. If the data file name cannot meet the requirements, it is recommended to use the file pattern to perform the import task. Otherwise, the import task can not scan the data files you need to import.
```

> - The data files and the schema files for building the database and tables must conform to the naming conventions.
> - If you can not provide the schema file for building the database, TiDB Cloud will create a target database for you according to the default configuration.
> - If you can not provide the schema file for building tables, you need to create the target tables in TiDB Cloud in advance.
> - If you can not provide data files with the required file names, you can modify the file pattern to perform the import task. Otherwise, TiDB Cloud can not scan the files you want to import.

Author: ```suggestion
- Prepare the SQL file for creating the target database (optional) and the SQL file for creating the target table that conform to the naming conventions. If the SQL file for creating the target table is not provided, you need to create the corresponding table manually in the target database in advance.
- Prepare a data file that conforms to the naming conventions for importing data. If the data file name cannot meet the requirements, it is recommended to use the file pattern to perform the import task. Otherwise, the import task can not scan the data files you need to import.
```

- Prepare the SQL file for creating the target database (optional) and the SQL file for creating the target table that conform to the naming conventions. If the SQL file for creating the target table is not provided, you need to create the corresponding table manually in the target database in advance.
- Prepare a data file that conforms to the naming conventions for importing data. If the data file name cannot meet the requirements, it is recommended to use the file pattern to perform the import task. Otherwise, the import task can not scan the data files you need to import.

## Naming conventions for schema files

This section describes the naming conventions for database and table schema files. The naming conventions for schema files are the same for all the following types of source files: CSV, Parquet, Aurora Snapshot, and SQL.

The naming conventions for schema files are as follows:

- Database schema file(optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`

> 

Author: @Frank945946 Please add some examples.

- Prepare the SQL file for creating the target database (optional) and the SQL file for creating the target table that conforms to the naming conventions. If the SQL file for creating the target table is not provided, you need to create the corresponding table manually in the target database in advance.
- Prepare a data file that conforms to the naming conventions for importing data. If the data file name can not meet the requirements, it is recommended to use the file pattern to perform the import task. Otherwise, the import task can not scan the data files you want to import.

## Naming conventions for schema files

This section describes the naming conventions for database and table schema files. The naming conventions for schema files are the same for all the following types of source files: CSV, Parquet, Aurora Snapshot, and SQL.

The naming conventions for schema files are as follows:

- Database schema file (optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`

For example:

> - `mytestdb-schema-create.sql`
> - `mytestdb.testtable-schema.sql`

Author: ```suggestion
- `import_db-schema-create.sql`
- `import_db.test_table-schema.sql`
```

## Naming conventions for data files

This section describes the naming conventions for data files. Depending on the type of source files, the naming conventions for data files are different.

### CSV

When you import CSV files, name the data files as follows:

- `${db_name}.${table_name}[.XXXXXX].csv` ([.XXXXXX] is optional)

For example:

- `import_db.test_table.csv`
- `import_db.test_table.01.csv`

### Parquet

When you import Parquet files, name the data files as follows:

- `${db_name}.${table_name}[.XXXXXX][.{snappy|gz|lzo}].parquet` (`[.XXXXXXX]` and `[.{snappy|gz|lzo}]` are optional)

For example:

- `import_db.test_table.parquet`
- `import_db.test_table.01.parquet`
- `import_db.test_table.gz.parquet`
- `import_db.test_table.01.gz.parquet`

### Aurora Snapshot

For Aurora Snapshot files, all files with the `.parquet` suffix in the `${db_name}.${table_name}/` conform to the naming convention. A data file name can contain any prefix consisting of "a-z, 0-9, - , _ , ." and suffix ".parquet".

For example:

> - `dbname.tablename.01.parquet`

Author: ```suggestion
- `import_db.test_table/mydata.parquet`
- `import_db.test_table/part001/mydata.parquet`
- `import_db.test_table/part002/mydata-part002.parquet`
```

- Prepare the SQL file for creating the target database (optional) and the SQL file for creating the target table that conforms to the naming conventions. If the SQL file for creating the target table is not provided, you need to create the corresponding table manually in the target database in advance.
- Prepare a data file that conforms to the naming conventions for importing data. If the data file name cannot meet the requirements, it is recommended to use the file pattern to perform the import task. Otherwise, the import task can not scan the data files you want to import.

## Naming conventions for schema files

This section describes the naming conventions for database and table schema files. The naming conventions for schema files are the same for all the following types of source files: CSV, Parquet, Aurora Snapshot, and SQL.

The naming conventions for schema files are as follows:

- Database schema file(optional): `${db_name}-schema-create.sql`
- Table schema file: `${db_name}.${table_name}-schema.sql`

For example:

- mytestdb-schema-create.sql
- mytestdb.testtable-schema.sql

> 

Reviewer: 1. In the examples below, the db name is `import_db` and the table name is `test_table`.  Do we need to unify the name here also ?
2. Do we need to also give an example of the file content? Like this: 

Database schema file name: `import_db-schema-create.sql`
File content example: 
```sql
CREATE DATABASE import_db;
```

Table schema file name: `import_db.test_table-schema.sql`
File content example: 
```sql
CREATE TABLE test_table (
    id INTEGER PRIMARY KEY,
    val VARCHAR(255)
);
```

Author: Good suggestion. Updated.

## Naming conventions for data files

This section describes the naming conventions for data files. Depending on the type of source files, the naming conventions for data files are different.

### CSV

When you import CSV files, name the data files as follows:

- `${db_name}.${table_name}[.XXXXXX].csv` ([.XXXXXX] is optional)

For example:

- `import_db.test_table.csv`
- `import_db.test_table.01.csv`

### Parquet

When you import Parquet files, name the data files as follows:

- `${db_name}.${table_name}[.XXXXXX][.{snappy|gz|lzo}].parquet` (`[.XXXXXXX]` and `[.{snappy|gz|lzo}]` are optional)

For example:

- `import_db.test_table.parquet`
- `import_db.test_table.01.parquet`
- `import_db.test_table.gz.parquet`
- `import_db.test_table.01.gz.parquet`

### Aurora Snapshot

When you import Aurora Snapshot files, name the data files as follows:

- All files with the `.parquet` suffix in the `${db_name}.${table_name}/` folder.

A data file name can contain any prefix consisting of "a-z, 0-9, - , _ , ." and suffix ".parquet". For example:

> - `dbname.tablename.01.parquet`

Reviewer: Here the naming example is too similar to other data formats.  What about these examples: 
- `import_db.test_table/mydata.parquet`
- `import_db.test_table/part001/mydata.parquet`
- `import_db.test_table/part002/mydata-part002.parquet`

The following is an example of a database schema file:

- Name: `import_db-schema-create.sql`
- File content:

    ```sql
    CREATE DATABASE import_db;
    ```

The following is an example of a table schema file:

- Name: `import_db.test_table-schema.sql`
- File content:

    ```sql
    CREATE TABLE test_table (
        id INTEGER PRIMARY KEY,
        val VARCHAR(255)
    );
    ```

## Naming conventions for data files

This section describes the naming conventions for data files. Depending on the type of source files, the naming conventions for data files are different.

### CSV

When you import CSV files, name the data files as follows:

- `${db_name}.${table_name}[.XXXXXX].csv` ([.XXXXXX] is optional)

For example:

- `import_db.test_table.csv`
- `import_db.test_table.01.csv`

### Parquet

When you import Parquet files, name the data files as follows:

- `${db_name}.${table_name}[.XXXXXX][.{snappy|gz|lzo}].parquet` (`[.XXXXXXX]` and `[.{snappy|gz|lzo}]` are optional)

For example:

- `import_db.test_table.parquet`
- `import_db.test_table.01.parquet`
- `import_db.test_table.gz.parquet`
- `import_db.test_table.01.gz.parquet`

### Aurora Snapshot

> For Aurora Snapshot files, all files with the `.parquet` suffix in the `${db_name}.${table_name}/` conform to the naming convention. A data file name can contain any prefix consisting of "a-z, 0-9, - , _ , ." and suffix ".parquet".

Reviewer 3: ```suggestion
For Aurora Snapshot files, all files with the `.parquet` suffix in the `${db_name}.${table_name}/` folder conform to the naming convention. A data file name can contain any prefix consisting of "a-z, 0-9, - , _ , ." and suffix ".parquet".
```
